metrics/src/main/java/org/aerogear/mobile/metrics/MetricsRegistry.java

```diff
@@ -0,0 +1,35 @@
+package org.aerogear.mobile.metrics;
+
+import org.aerogear.mobile.metrics.interfaces.MetricsProvider;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The metrics registry is the central storage for all metrics providers
+ */
+public final class MetricsRegistry {
+    private static MetricsRegistry INSTANCE;
+
+    private final Set<MetricsProvider> providers = new HashSet<>();
+
+    private MetricsRegistry() {
+        INSTANCE = this;
+    }
+
+    public void registerProvider(MetricsProvider provider) {
+        providers.add(provider);
+    }
+
+    public Set<MetricsProvider> getProviders() {
+        return this.providers;
+    }
+
+    public static MetricsRegistry instance() {
```

Comment:
Can we get dependency injection over singletons?

Reply:
I'd prefer that too. But in that case we would need dependency injection across multiple modules. Will have a look at dagger2, maybe it can help us here.

Reply:
We been looking at dagger2 - investigation was closed as it was not possible to use it on library level.